third_party_integration/huggingface_transformers/BootstrapNAS_HF/README.md

```diff
@@ -0,0 +1,82 @@
+# Integrating BootstrapNAS into Transformers
+https://github.com/huggingface/transformers
+
+This folder contains a git patch to enable NNCF-based neural architecture search for XNLI, SQuAD and GLUE training pipelines of the huggingface transformers repository. 
+
+Instructions:
+1. Apply the `0001-Modifications-for-bootstrapNAS-usage.patch` file to the huggingface transformers repository checked out at commit id: `bd469c40659ce76c81f69c7726759d249b4aef49`
+
+2. Install the `transformers` library and the example scripts from the patched repository as described in the documentation for the huggingface transformers repository.
+
+3. To start neural architecture search of NLP models using BootstrapNAS, use the regular scripts and command line parameters for training, but with additional `--nncf_config <path_to_nncf_config>` parameter.
+The NNCF configs to be used in this way are also provided in the same patch on a per-model basis.
+
+
+## Training Script (Trained on 1x Tesla V100)
+
+
+### BERT-XNLI
+
+    python examples/pytorch/text-classification/run_xnli.py \
+        --model_name_or_path bert-base-chinese \
+        --language zh \
+        --train_language zh \
+        --do_train \
+        --do_eval \
+        --do_search \
+        --per_gpu_train_batch_size 48 \
+        --per_gpu_eval_batch_size 64 \
+        --max_seq_length 128 \
+        --output_dir results/bert_xnli_nas \
+        --nncf_config nncf_bootstrapnas_config/nncf_bootstrapnas_bert_config_xnli.json \
+        --evaluation_strategy epoch \
+        --save_strategy epoch \
+        --metric_for_best_model accuracy
+
+### BERT-SQuAD v1.1
+
+
+    python examples/pytorch/question-answering/run_qa.py \
```

Comment:
We can't make merge without automatic verification that it's working.
Propose at least extend sanity tests for the HF patch, like it was done in JPQD: https://github.com/openvinotoolkit/nncf/pull/1319/files#diff-7e1e1fc22e5328780e24fc6938d6cca183b4729d6c820a48ad9d8824247323ceR40

If there are configs to reproduce some results, it should be added to nightly/weekly testing:
https://github.com/openvinotoolkit/nncf/pull/1319/files#diff-612fbad66326ad88318637d262bc83de99f7ffcdbdd9c145ef1c293f65ae0f1eR302

Reply:
Added tests. Thanks!